kolibri_explore_plugin/assets/src/apiResources.js

```diff
@@ -36,6 +38,28 @@ export const ContentNodeResource = new Resource({
   fetchNextSteps(getParams) {
     return this.fetchDetailCollection('next_steps', Store.getters.currentUserId, getParams);
   },
+  fetchChannelAsync(channelId, parent = null) {
+    // Fetch all content for a channel recursively, making a request to the
+    // backend for each level in the nodes tree
+    const getParams = {
+      channel_id: channelId,
+      parent: parent || channelId,
+    };
+
+    return this.fetchCollection({ getParams }).then(n => {
```

Comment:
There must be a beter way to fetch all topic nodes of a parent node. Directly, without fetching all and then filtering. Maybe something like this that I just found in `kolibri/plugins/coach/assets/src/modules/lessonResources/handlers.js`:

```
    return ContentNodeSearchResource.fetchCollection({
      getParams: {
        search: params.searchTerm,
        ...pickBy({
          kind: query.kind,
          channel_id: query.channel,
        }),
      },
```

fetchDescendants

Reply:
The `fetchDescendants` doesn't return all the node attributes, and if we filter by kind we won't get the content nodes that we're returning.

I played a bit with a custom filter in the `viewsets.py`, but at the end, if we get all the content in one query, there's an exception in the consolidate method, when it tries to generate the `ancestors`. We can even try to fix this on our override of the consolidate, but we'll loose the `ancestors` that's used in some places.

Reply:
OK, sounds good then. Thanks for having taken an extra look.

Reply:
And sorry, the `fetchDescendants` was a leftover, it wasn't intended to be part of my comment. I copy/pasted while digging the source code.